packages/core/src/rules/common/no-required-schema-properties-undefined.ts

```diff
@@ -14,27 +16,25 @@ export const NoRequiredSchemaPropertiesUndefined:
   | Arazzo1Rule = () => {
   return {
     Schema: {
-      enter(
-        schema: Oas3Schema | Oas3_1Schema | Oas2Schema,
-        { location, report, resolve }: UserContext
-      ) {
+      enter(schema: AnySchema, { location, report, resolve, parents }: UserContext) {
         if (!schema.required) return;
```

Comment:
You can collect and drop the parent directly inside the visitor with something like this:

```suggestion
const parents = []
return {
Schema: {
      leave() {
        parents.pop(); 
      },
      enter(schema: AnySchema, { location, report, resolve, parents }: UserContext) {
if (!schema.required) { 
  // somewhere here we can check if it's inside `oneOf`/`allOf`
  parents.push(schema); // do the same in the bottom of this function
  return;
}
```